README.md

```diff
@@ -60,3 +62,44 @@ Then use the `analyticsClient` variable:
+Reach out to the data team when in doubt which project to use or if you need to define a new project.
+
+In the final destinations, we separate staging and production environments. We use the same `projectName` value but different tokens to differentiate the production and staging environments. So you only need to update `endpoint` to pick the right destination.
+
+Use the staging environment for testing purposes by setting `endpoint` in the `createClient` initialization to `data.balena-staging.com`. Once you are happy with your tests and ready to go live, update the `endpoint` to `data.balena-cloud.com` for production environment.
+
+Additionally, identify your component using the parameter `componentName`. It's best to keep the value short yet descriptive.
+
+Please ask Operation folks for `#access` to the relevant project in your final destination so you can validate that the tracked events show as expected.
+
+Finally, there is the optional configuration parameter `componentVersion` to track the version of a component sending each event. This is useful to track changes across new component versions so we strongly recommend tracking it.
+
+## Web tracker
+
+[`WebTracker`](src/web.ts) exposes an events tracking interface suitable for web apps. It provides a simple `Page View` event tracking with page loading metrics via `trackPageView`.
+
+When initializing `createWebTracker`, set `prefix` as a short string that describes your project. All events tracked by `WebTracker` will have this `prefix` in their name.
+We use this prefixing method to identify and group events in the final destination for organization purposes and helping the analysts traverse the events with ease.
+This is especially useful when there are multiple components sending events to the same project in the final destination. To give an example, our marketing website, docs pages and blog all send `Page View` event but these are prefixed respectively `Marketing`, `Docs` and `Blog`. So the final event name is `[Marketing] Page View`, `[Docs] Page View`, etc. Furthermore, this method makes it easier to identify all the `Marketing` events.
+
+## Identifying users
+
+By default the client tracks `deviceId` to identify the current device of the end user. `deviceId` is a UUID generated automatically by the client. Use `setUserId` method to identify authenticated user. At least one of these identifiers is required to successfully track an event.
+
+In most cases the device refers to the browser of the user (or Electron application for the case of Etcher). By persisting the `deviceId` in the local storage, we are able to identify the events coming from the same user in order to create a timeline of component usage.
+
+It is possible to generate a new device identifier and also override an existing device identifier. We use these methods especially when we need to link multiple device identifiers that we know belonging to the same user.
```

Comment:
Why is this needed? My impression was that different devices can belong to the same user - thus device ID and user ID are accounted for. Or is it actually the same thing? :)

Reply:
Looking into the code more, it's probably about updating the collection of device IDs rather than overriding a single ID, is that right?

Reply:
Yes, you captured the common cases. The next paragraphs also expand on these.

There are some more to it: as we transition some components from Mixpanel to Amplitude, we prefer to re-use the Mixpanel identifiers as Amplitude identifiers (not to restart the events timeline of existing user). Also in the case of unauthenticated usage as in Etcher, we rely on `deviceId` to track the same user. So a component maintainer might need to use these methods mentioned to achieve this.

Shall we add anything extra to this paragraph?

Reply:
> we prefer to re-use the Mixpanel identifiers as Amplitude identifiers (not to restart the events timeline of existing user)

This is perfectly understandable, but does it involve overwriting the *device* ID specifically?

Reply:
> does it involve overwriting the device ID specifically?

The answer depends on the setup. 

Now I see how this sentence is confusing. I'm pushing a reworked version Pranas, let me know if it's still confusing.